assume/units/steel_plant.py

```diff
@@ -17,29 +17,13 @@
 from assume.common.market_objects import MarketConfig, Orderbook
 from assume.common.utils import get_products_index
 from assume.units.dsm_load_shift import DSMFlex
-from assume.units.dst_components import (
-    create_driplant,
-    create_dristorage,
-    create_electric_arc_furnance,
-    create_electrolyser,
-    create_hydrogen_storage,
-)
 
 SOLVERS = ["gurobi", "glpk", "cbc", "cplex"]
 
 logger = logging.getLogger(__name__)
 
-# Mapping of component type identifiers to their respective classes
```

Comment:
@nick-harder The original idea was to use this dictionary of technologies to be unique to each plant type or demand-side agent (e.g., only technologies compatible with a steel plant, cement plant, or buildings). The idea is to exclude technologies that aren't relevant to the specific agent types, ensuring that only compatible technologies are modeled for each.

Reply:
@Manish-Khanra I moved this function to the common class since it is where it belongs. But you are right, we need to have a check inside the steel plant itself for the components provided in the component list, if it makes sense or not.

Reply:
@Manish-Khanra I have now included required_technologies and optional_technologies to the steel plant so that the components are now checked to containt all required ones and not contain any extra ones.